e2e/rbd.go

```diff
@@ -334,6 +335,229 @@ var _ = Describe("RBD", func() {
+				err = waitForDaemonSets(rbdDaemonsetName, cephCSINamespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					e2elog.Failf("timeout waiting for daemonset pods with error %v", err)
+				}
+
+				opt := metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("app=%s", app.Name),
+				}
+				// Write something to mountpoint and expect a failure
+				filePath := app.Spec.Containers[0].VolumeMounts[0].MountPath + "/test"
+				_, stdErr := execCommandInPodAndAllowFail(f, fmt.Sprintf("echo 'Hello World' > %s", filePath), app.Namespace, &opt)
+				IOErr := fmt.Sprintf("cannot create %s: Input/output error", filePath)
+				if !strings.Contains(stdErr, IOErr) {
+					e2elog.Failf(stdErr)
+				} else {
+					e2elog.Logf("failed IO as expected: %v", stdErr)
+				}
+
+				err = deletePVCAndApp("", f, pvc, app)
+				if err != nil {
+					e2elog.Failf("failed to delete PVC and application with error %v", err)
+				}
+				// validate created backend rbd images
+				validateRBDImageCount(f, 0)
+			})
+
+			By("restart rbd-nbd process on nodeplugin and continue IO after nodeplugin restart", func() {
+				err := deleteResource(rbdExamplePath + "storageclass.yaml")
+				if err != nil {
+					e2elog.Failf("failed to delete storageclass with error %v", err)
+				}
+				// Tweak Storageclass to add netlink,reattach rbd-nbd mounter options
+				scOpts := map[string]string{
+					"mounter":    "rbd-nbd",
+					"mapOptions": "try-netlink,reattach-timeout=180",
+				}
+				err = createRBDStorageClass(f.ClientSet, f, nil, scOpts, deletePolicy)
+				if err != nil {
+					e2elog.Failf("failed to create storageclass with error %v", err)
+				}
+
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC with error %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+
+				app, err := loadApp(appPath)
+				if err != nil {
+					e2elog.Failf("failed to load application with error %v", err)
+				}
+				app.Namespace = f.UniqueName
+				label := map[string]string{
+					"app": app.Name,
+				}
+				app.Labels = label
+				app.Spec.Volumes[0].PersistentVolumeClaim.ClaimName = pvc.Name
+				err = createPVCAndApp("", f, pvc, app, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC and application with error %v", err)
+				}
+				// validate created backend rbd images
+				validateRBDImageCount(f, 1)
+
+				// Get details of rbd-nbd process
+				// # ps -eo 'cmd' | grep [r]bd-nbd
+				// /usr/bin/rbd-nbd --id cephcsi-rbd-node -m svc-name:6789 --keyfile=/tmp/csi/keys/keyfile attach --device /dev/nbd0 pool-name/image-name --try-netlink --reattach-timeout=180
+				opt1 := metav1.ListOptions{
+					LabelSelector: "app in (ceph-csi-rbd, csi-rbdplugin)",
+				}
+				uname, stdErr, err := execCommandInContainer(f, "uname -a", cephCSINamespace, "csi-rbdplugin", &opt1)
+				if err != nil || stdErr != "" {
+					e2elog.Failf("failed to run uname cmd : %v, stdErr: %v ", err, stdErr)
+				}
+				e2elog.Logf("uname 1: %v", uname)
```

Comment:
@pkalever instead of `1,2,..etc` may be we can log other marker?

Reply:
Humble, right now, the helm chart e2e tests are failing for rbd-nbd, so I wanted to debug, hence added some extra debug logs. These logs will not be part of the original patch and I shall remove them once everything runs smooth.

Thanks for keeping an eye on this PR!